bin/zbm_kcl

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+zbm_load_lib() {
+  lib="${1?no library specified for load}"
+
+  : "${ZBM_MODULEDIR:=/usr/lib/dracut/modules.d/90zfsbootmenu}"
+
+  if [ -r "${ZBM_MODULEDIR}/lib/${lib}" ]; then
+    # shellcheck disable=SC1090
+    source "${ZBM_MODULEDIR}/lib/${lib}"
+  else
+    echo "ERROR: ${ZBM_MODULEDIR}/lib/${lib} not found" >&2
+    echo "Set ZBM_MODULEDIR to the root of the ZBM dracut module" >&2
+    exit 1
+  fi
+}
+
+usage() {
+  cat <<-EOF
+	USAGE: $0 [-a <arg>] [-r <arg>] [-e] [-d] [-v] [bootenv]
+        Review or update kernel command line (KCL) associated with <bootenv>
+        When the boot environment is specified, the current root will be used
+
+	-a <arg>: Append an argument
+	-r <arg>: Remove an argument
+	-e: Open the KCL for editing in \$EDITOR
+	-d: Remove entire command line
+	-v: Enable verbose output
+	EOF
+}
+
+strip_kcl() {
+  awk '
+  BEGIN { first = 1; }
+
+  /^$/ { exit; }
+
+  {
+    gsub("[[:space:]]*#.*", "");
+
+    if (length == 0) next;
+
+    if (first == 1) {
+      first = 0;
+    } else {
+      printf " ";
+    }
+
+    printf "%s", $0;
+  }
+  '
+}
+
+delete() {
+  local bootenv="${1}"
+
+  if ! zfs list -o name -H "${bootenv}" >/dev/null 2>&1; then
+    zerror "no boot environment specified"
+    usage
+    return 1
+  fi
+
+  if ! zfs inherit org.zfsbootmenu:commandline "${bootenv}"; then
+    zerror "failed to remove KCL from ${bootenv}"
+    return 1
+  fi
+}
+
+raw_kcl() {
+  local bootenv="${1}"
+  local kcl
+
+  kcl="$(zfs get -H -o value org.zfsbootmenu:commandline "${bootenv}")" || return 1
+  [ "${kcl}" = "-" ] && return
+
+  echo "${kcl}"
+}
+
+review() {
+  local bootenv="${1}"
+  local kcl
+
+  kcl="$(raw_kcl "${bootenv}")"
+  echo "${kcl}"
+
+  if [[ "${kcl}" =~ "%{parent}" ]]; then
+    echo "# With parent references, KCL expands to"
+    echo "#  $(read_kcl_prop "${bootenv}")"
+  fi
+
+  return 0
+}
+
+edit() {
+  local bootenv kclfile oldsum newkcl newsum
+
+  bootenv="${1}"
+  kclfile="$(mktemp)"
+
+  trap 'rm -f "${kclfile}"' RETURN
+
+  if [ -z "${EDITOR}" ]; then
```

Comment:
This conflicts with line 125. Can it default to `vi` here if undefined, and then only exit if `vi` isn't in $PATH ?

Reply:
I meant to delete that when setting the default later, but I've now improved it to default to `vi` and fail early when `${EDITOR}` can't be found.